Assets/Tests/PlayMode/EOSSessionsManagerTests.cs

```diff
@@ -0,0 +1,654 @@
+            attribute.Key = "Level";
+            attribute.AsString = "UNITYTESTS";
+            attribute.ValueType = AttributeType.String;
+            attribute.Advertisement = SessionAttributeAdvertisementType.Advertise;
+
+            sessionCreationParameters.Attributes.Add(attribute);
+
+            return sessionCreationParameters;
+        }
+
+        /// <summary>
+        /// Generates a random Session name that is compliant with the rules.
+        /// This should be random in the odd case that a Session is not properly cleaned up between tests.
+        /// </summary>
+        /// <returns>A random name.</returns>
+        private string GetRandomSessionName()
+        {
+            const int numberOfCharacters = 10;
+            StringBuilder randomName = new StringBuilder();
+
+            for (int createdSessionIndex = 0; createdSessionIndex < numberOfCharacters; createdSessionIndex++)
+            {
+                randomName.Append(UnityEngine.Random.Range(0, 10).ToString());
+            }
+
+            return randomName.ToString();
+        }
+
+        /// <summary>
+        /// Helper function for creating a Session for tests.
+        /// </summary>
+        /// <param name="toCreate">Parameters of the Session to create.</param>
+        /// <param name="onComplete">Action to run with result.</param>
+        /// <param name="presence">Indicates the Session should be created as a Presence Session.</param>
+        /// <returns>Yieldable instruction that finishes when the Session is created.</returns>
+        private IEnumerator CreateSession(Session toCreate, Action<Result> onComplete, bool presence)
+        {
+            const float MostTimeToWaitForSessionCreation = 5f;
+
+            string resultingCreationSessionName = null;
+            Result? resultingCreationResult = null;
+
+            UnityAction<string, Result> handleCreationResult = (string relevantSession, Result relevantResult) =>
+            {
+                resultingCreationSessionName = relevantSession;
+                resultingCreationResult = relevantResult;
+            };
+
+            ManagerInstance.UIOnSessionCreated.AddListener(handleCreationResult);
+
+            ManagerInstance.CreateSession(toCreate, presence);
+
+            yield return new WaitUntilDone(MostTimeToWaitForSessionCreation, () => resultingCreationResult.HasValue);
+
+            ManagerInstance.UIOnSessionCreated.RemoveListener(handleCreationResult);
+
+            onComplete.Invoke(resultingCreationResult.Value);
+        }
+
+        private IEnumerator WaitWhileUpdatingManager(float secondsToWait)
+        {
+            float startTime = Time.realtimeSinceStartup;
+            float endTime = startTime + secondsToWait;
+
+            while (Time.realtimeSinceStartup < endTime)
+            {
+                ManagerInstance.Update();
+                yield return new WaitForEndOfFrame();
+            }
+        }
```

Comment:
This function does not appear to be referenced anywhere.

Reply:
Resolved by removing this function. It can be made again if it matters to.